### PR TITLE
fix(cd): Fix workspace reference to allow for automatic versioning

### PR DESCRIPTION
**Pull request by zFernand0**
## Proposed changes

- Replace the old repository tokens with the new zowe org-wide tokens
- Replace the "workspace:" protocol to use the package path instead of the independent version.
  Reasons:
    - Lerna won't touch this protocol when using `pnpm` ([source](https://lerna.js.org/docs/recipes/using-pnpm-with-lerna))
    - Running a `pnpm install` after a `lerna version` will cause the following error
        ```
         ERR_PNPM_NO_MATCHING_VERSION_INSIDE_WORKSPACE  In : No matching version found for 
              eslint-plugin-zowe-explorer@workspace:3.0.0-next.202309141150 inside the workspace
        ```
    - Decided on `./packages/eslint...` vs `packages/eslint...` because the latter is considered invalid. See error below:
        ```
         ERROR  Invalid workspace: spec (workspace:packages/eslint-plugin-zowe-explorer)
        ```

## Release Notes

Milestone: N/A

Changelog: Update CD workflow to support publishing with PNPM

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [ ] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

- Further testing is being done in this fork: https://github.com/zFernand0/vscode-extension-for-zowe/pull/7
- This PR depends on: https://github.com/zowe-actions/octorelease/pull/116

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "eslint": "^8.34.0",
     "eslint-config-prettier": "^8.6.0",
     "eslint-plugin-prettier": "^4.2.1",
-    "eslint-plugin-zowe-explorer": "workspace:3.0.0-next.202309141150",
```
Comment (JillieBeanSim) [inline package.json:32]: we only have to do this with the ZE eslint package? not the ZE API package too in dependencies?

**Review by JillieBeanSim — COMMENTED** [review]
Does the ZE package.json with this dependency need updated too and will we need this for ZE API dependency in other package.jsons?
`"eslint-plugin-zowe-explorer": "workspace:./packages/eslint-plugin-zowe-explorer"`

**Comment by zFernand0** [comment]
> Does the ZE package.json with this dependency need updated too and will we need this for ZE API dependency in other package.jsons?
> `"eslint-plugin-zowe-explorer": "workspace:./packages/eslint-plugin-zowe-explorer"`
> Screen Shot 2023-10-20 at 12 47 45 PM
> 
> 

Not necessarily. The individual packages can keep the version numbers as the `lerna version` command can successfully update them. This was needed for the root version since we need to dedupe the dependency, but lerna won't update the root of the project since we don't treat the root as a package in the workspace.

**Review by traeok — APPROVED** [review]
LGTM, thanks @zFernand0 😁

**Pull request merged**